diff -ur giflib-4.1.6.orig/util/gif2epsn.c giflib-4.1.6/util/gif2epsn.c
--- giflib-4.1.6.orig/util/gif2epsn.c	2007-11-10 19:46:31.000000000 +0100
+++ giflib-4.1.6/util/gif2epsn.c	2012-12-12 13:25:38.930805507 +0100
@@ -146,7 +146,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gif2ps.c giflib-4.1.6/util/gif2ps.c
--- giflib-4.1.6.orig/util/gif2ps.c	2007-11-10 19:46:31.000000000 +0100
+++ giflib-4.1.6/util/gif2ps.c	2012-12-12 13:25:50.657471760 +0100
@@ -129,7 +129,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gif2rgb.c giflib-4.1.6/util/gif2rgb.c
--- giflib-4.1.6.orig/util/gif2rgb.c	2007-11-10 19:46:31.000000000 +0100
+++ giflib-4.1.6/util/gif2rgb.c	2012-12-12 13:26:03.160804652 +0100
@@ -108,7 +108,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gif2rle.c giflib-4.1.6/util/gif2rle.c
--- giflib-4.1.6.orig/util/gif2rle.c	2012-12-12 13:24:53.480807110 +0100
+++ giflib-4.1.6/util/gif2rle.c	2012-12-12 13:28:54.604131937 +0100
@@ -103,7 +103,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gif2x11.c giflib-4.1.6/util/gif2x11.c
--- giflib-4.1.6.orig/util/gif2x11.c	2007-11-10 19:46:31.000000000 +0100
+++ giflib-4.1.6/util/gif2x11.c	2012-12-12 13:29:00.424131733 +0100
@@ -143,7 +143,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifasm.c giflib-4.1.6/util/gifasm.c
--- giflib-4.1.6.orig/util/gifasm.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifasm.c	2012-12-12 13:26:18.684137438 +0100
@@ -98,7 +98,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifbg.c giflib-4.1.6/util/gifbg.c
--- giflib-4.1.6.orig/util/gifbg.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifbg.c	2012-12-12 13:26:29.064137071 +0100
@@ -130,7 +130,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifclip.c giflib-4.1.6/util/gifclip.c
--- giflib-4.1.6.orig/util/gifclip.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifclip.c	2012-12-12 13:26:47.164136433 +0100
@@ -92,7 +92,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifclrmp.c giflib-4.1.6/util/gifclrmp.c
--- giflib-4.1.6.orig/util/gifclrmp.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifclrmp.c	2012-12-12 13:28:42.594132361 +0100
@@ -111,7 +111,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifcolor.c giflib-4.1.6/util/gifcolor.c
--- giflib-4.1.6.orig/util/gifcolor.c	2005-10-10 08:22:22.000000000 +0200
+++ giflib-4.1.6/util/gifcolor.c	2012-12-12 13:28:35.667465940 +0100
@@ -89,7 +89,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifcomb.c giflib-4.1.6/util/gifcomb.c
--- giflib-4.1.6.orig/util/gifcomb.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifcomb.c	2012-12-12 13:26:55.544136137 +0100
@@ -93,7 +93,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/giffix.c giflib-4.1.6/util/giffix.c
--- giflib-4.1.6.orig/util/giffix.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/giffix.c	2012-12-12 13:27:03.770802514 +0100
@@ -90,7 +90,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifflip.c giflib-4.1.6/util/gifflip.c
--- giflib-4.1.6.orig/util/gifflip.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifflip.c	2012-12-12 13:27:30.344134910 +0100
@@ -104,7 +104,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifhisto.c giflib-4.1.6/util/gifhisto.c
--- giflib-4.1.6.orig/util/gifhisto.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifhisto.c	2012-12-12 13:27:34.234134772 +0100
@@ -112,7 +112,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifinfo.c giflib-4.1.6/util/gifinfo.c
--- giflib-4.1.6.orig/util/gifinfo.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifinfo.c	2012-12-12 13:27:38.400801292 +0100
@@ -104,7 +104,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	printf("Format: quoted text string used for formating of information\n");
 	printf("  Special characters for various information can be used\n");
diff -ur giflib-4.1.6.orig/util/gifinter.c giflib-4.1.6/util/gifinter.c
--- giflib-4.1.6.orig/util/gifinter.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifinter.c	2012-12-12 13:27:41.084134531 +0100
@@ -98,7 +98,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifinto.c giflib-4.1.6/util/gifinto.c
--- giflib-4.1.6.orig/util/gifinto.c	2005-10-10 08:22:22.000000000 +0200
+++ giflib-4.1.6/util/gifinto.c	2012-12-12 13:27:43.884134432 +0100
@@ -96,7 +96,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifovly.c giflib-4.1.6/util/gifovly.c
--- giflib-4.1.6.orig/util/gifovly.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifovly.c	2012-12-12 13:27:46.730800998 +0100
@@ -73,7 +73,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifpos.c giflib-4.1.6/util/gifpos.c
--- giflib-4.1.6.orig/util/gifpos.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifpos.c	2012-12-12 13:27:49.477467568 +0100
@@ -90,7 +90,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifrotat.c giflib-4.1.6/util/gifrotat.c
--- giflib-4.1.6.orig/util/gifrotat.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifrotat.c	2012-12-12 13:27:51.697467490 +0100
@@ -117,7 +117,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifrsize.c giflib-4.1.6/util/gifrsize.c
--- giflib-4.1.6.orig/util/gifrsize.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifrsize.c	2012-12-12 13:27:54.127467404 +0100
@@ -113,7 +113,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/giftext.c giflib-4.1.6/util/giftext.c
--- giflib-4.1.6.orig/util/giftext.c	2007-11-10 05:28:56.000000000 +0100
+++ giflib-4.1.6/util/giftext.c	2012-12-12 13:28:03.927467058 +0100
@@ -102,7 +102,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/gifwedge.c giflib-4.1.6/util/gifwedge.c
--- giflib-4.1.6.orig/util/gifwedge.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/gifwedge.c	2012-12-12 13:28:06.577466965 +0100
@@ -94,7 +94,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/icon2gif.c giflib-4.1.6/util/icon2gif.c
--- giflib-4.1.6.orig/util/icon2gif.c	2005-10-10 08:22:22.000000000 +0200
+++ giflib-4.1.6/util/icon2gif.c	2012-12-12 13:28:09.410800198 +0100
@@ -94,7 +94,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/raw2gif.c giflib-4.1.6/util/raw2gif.c
--- giflib-4.1.6.orig/util/raw2gif.c	2005-10-10 08:22:22.000000000 +0200
+++ giflib-4.1.6/util/raw2gif.c	2012-12-12 13:28:12.040800106 +0100
@@ -118,7 +118,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/rgb2gif.c giflib-4.1.6/util/rgb2gif.c
--- giflib-4.1.6.orig/util/rgb2gif.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/rgb2gif.c	2012-12-12 13:28:14.757466676 +0100
@@ -111,7 +111,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/rle2gif.c giflib-4.1.6/util/rle2gif.c
--- giflib-4.1.6.orig/util/rle2gif.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/rle2gif.c	2012-12-12 13:28:18.377466549 +0100
@@ -105,7 +105,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
diff -ur giflib-4.1.6.orig/util/text2gif.c giflib-4.1.6/util/text2gif.c
--- giflib-4.1.6.orig/util/text2gif.c	2005-10-10 08:22:23.000000000 +0200
+++ giflib-4.1.6/util/text2gif.c	2012-12-12 13:28:24.270799674 +0100
@@ -107,7 +107,7 @@
     }
 
     if (HelpFlag) {
-	fprintf(stderr, VersionStr);
+	fprintf(stderr, "%s", VersionStr);
 	GAPrintHowTo(CtrlStr);
 	exit(EXIT_SUCCESS);
     }
