--- giflib-4.1.4/configure.ac~orig	2008-11-04 23:06:29.000000000 +0100
+++ giflib-4.1.4/configure.ac	2008-11-04 23:07:37.000000000 +0100
@@ -31,9 +31,12 @@
                              AC_DEFINE(HAVE_LIBGL_S, [1], [Define if the gl_s library is installed (SGI GL library)])
                  COMPILABLE_EXTRAS="${COMPILABLE_EXTRAS} gif2iris"],, $X_LIBS)
 
+    X11_LIB=""
+if test "x$with_x" != xno; then
     AC_CHECK_LIB(X11, main, [X11_LIB="${X11_LIB} $X_LIBS ${X_EXTRA_LIBS} -lX11"
                              AC_DEFINE(HAVE_LIBX11, [1], [Define if the X11 library is installed])
                  COMPILABLE_EXTRAS="${COMPILABLE_EXTRAS} gif2x11"],, $X_LIBS)
+fi
     DEVS="${GL_S_LIB} ${X11_LIB}" 
     AC_SUBST(DEVS)
     AC_SUBST(COMPILABLE_EXTRAS)
