--- giflib-4.1.6/configure.ac~	2009-04-08 14:28:01.000000000 +0200
+++ giflib-4.1.6/configure.ac	2009-04-08 14:29:23.000000000 +0200
@@ -54,7 +54,7 @@
 if test "x$enable_x11" != "xno" ; then
   AC_PATH_XTRA
   AC_CHECK_HEADERS([X11/Xlib.h X11/Xutil.h],
-    AC_CHECK_LIB(X11, main, [X11_LIB="${X11_LIB} ${X_PRE_LIBS} $X_LIBS ${X_EXTRA_LIBS} -lX11"
+    AC_CHECK_LIB(X11, main, [X11_LIB="${X11_LIB} $X_LIBS ${X_EXTRA_LIBS} -lX11"
                  AC_DEFINE(HAVE_LIBX11, [1], [Define if the X11 library is installed])
                  COMPILABLE_EXTRAS="${COMPILABLE_EXTRAS} gif2x11${EXEEXT}"],, $X_LIBS))
 fi
