--- giflib-5.1.8/Makefile.orig	2019-03-17 17:33:02.000000000 +0100
+++ giflib-5.1.8/Makefile	2019-03-27 18:05:08.720055399 +0100
@@ -58,15 +58,15 @@
 	gifsponge \
 	gifwedge
 
-LDLIBS=libgif.a -lm
+LDLIBS=-L. -lgif -lm
 
 all: libgif.so libgif.a $(UTILS)
 	$(MAKE) -C doc
 
-$(UTILS):: libgif.a
+$(UTILS):: libgif.so
 
 libgif.so: $(OBJECTS) $(HEADERS)
-	$(CC) $(CFLAGS) -shared $(OFLAGS) -Wl,-soname -Wl,libgif.so.$(LIBMAJOR) -o libgif.so $(OBJECTS)
+	$(CC) $(LDFLAGS) $(CFLAGS) -shared $(OFLAGS) -Wl,-soname -Wl,libgif.so.$(LIBMAJOR) -o libgif.so $(OBJECTS)
 
 libgif.a: $(OBJECTS) $(HEADERS)
 	$(AR) rcs libgif.a $(OBJECTS)
