--- giflib-6.1.3/Makefile.orig	2026-04-12 19:17:27.000000000 +0200
+++ giflib-6.1.3/Makefile	2026-05-17 16:14:28.886294920 +0200
@@ -75,7 +75,7 @@ UTILS = $(INSTALLABLE) \
 	gifwedge \
 	gif2rgb
 
-LDLIBS=libgif.a -lm
+LDLIBS=-L. -lgif -lm
 
 MANUAL_PAGES_1 = \
 	doc/gifbuild.xml \
@@ -95,19 +95,19 @@ SOEXTENSION	= so
 LIBGIFSO	= libgif.$(SOEXTENSION)
 LIBGIFSOMAJOR	= libgif.$(SOEXTENSION).$(LIBMAJOR)
 LIBGIFSOVER	= libgif.$(SOEXTENSION).$(LIBVER)
-LIBUTILSO	= libutil.$(SOEXTENSION)
-LIBUTILSOMAJOR	= libutil.$(SOEXTENSION).$(LIBMAJOR)
+LIBUTILSO	= libgifutil.$(SOEXTENSION)
+LIBUTILSOMAJOR	= libgifutil.$(SOEXTENSION).$(LIBMAJOR)
 ifeq ($(UNAME), Darwin)
 SOEXTENSION	= dylib
 LIBGIFSO        = libgif.$(SOEXTENSION)
 LIBGIFSOMAJOR   = libgif.$(LIBMAJOR).$(SOEXTENSION)
 LIBGIFSOVER	= libgif.$(LIBVER).$(SOEXTENSION)
-LIBUTILSO	= libutil.$(SOEXTENSION)
-LIBUTILSOMAJOR	= libutil.$(LIBMAJOR).$(SOEXTENSION)
+LIBUTILSO	= libgifutil.$(SOEXTENSION)
+LIBUTILSOMAJOR	= libgifutil.$(LIBMAJOR).$(SOEXTENSION)
 endif
 
 SHARED_LIBS = $(LIBGIFSO) $(LIBUTILSO)
-STATIC_LIBS = libgif.a libutil.a
+STATIC_LIBS = libgif.a libgifutil.a
 
 all: shared-lib static-lib $(UTILS)
 ifeq ($(UNAME), Darwin)
@@ -115,7 +115,7 @@ else
 	$(MAKE) -C doc
 endif
 
-$(UTILS):: $(STATIC_LIBS)
+$(UTILS):: $(SHARED_LIBS)
 
 shared-lib: $(SHARED_LIBS)
 
@@ -131,15 +131,15 @@ endif
 libgif.a: $(OBJECTS) $(HEADERS)
 	$(AR) rcs libgif.a $(OBJECTS)
 
-$(LIBUTILSO): $(UOBJECTS) $(UHEADERS)
+$(LIBUTILSO): $(UOBJECTS) $(UHEADERS) $(LIBGIFSO)
 ifeq ($(UNAME), Darwin)
 	$(CC) $(CFLAGS) -dynamiclib -current_version $(LIBVER) $(UOBJECTS) -o $(LIBUTILSO)
 else
-	$(CC) $(CFLAGS) $(CPPLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,$(LIBUTILSOMAJOR) -o $(LIBUTILSO) $(UOBJECTS)
+	$(CC) $(CFLAGS) $(CPPLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,$(LIBUTILSOMAJOR) -o $(LIBUTILSO) $(UOBJECTS) -L. -lgif
 endif
 
-libutil.a: $(UOBJECTS) $(UHEADERS)
-	$(AR) rcs libutil.a $(UOBJECTS)
+libgifutil.a: $(UOBJECTS) $(UHEADERS)
+	$(AR) rcs libgifutil.a $(UOBJECTS)
 
 clean:
 	rm -f $(UTILS) $(OBSOLETE_UTILS) $(TARGET) libgetarg.a $(SHARED_LIBS) $(STATIC_LIBS) *.o
@@ -178,14 +178,19 @@ install-bin: $(INSTALLABLE)
 install-include:
 	$(INSTALL) -d "$(DESTDIR)$(INCDIR)"
 	$(INSTALL) -m 644 gif_lib.h "$(DESTDIR)$(INCDIR)"
+	$(INSTALL) -m 644 getarg.h "$(DESTDIR)$(INCDIR)/gif_util.h"
 install-static-lib:
 	$(INSTALL) -d "$(DESTDIR)$(LIBDIR)"
 	$(INSTALL) -m 644 libgif.a "$(DESTDIR)$(LIBDIR)/libgif.a"
+	$(INSTALL) -m 644 libgifutil.a "$(DESTDIR)$(LIBDIR)/libgifutil.a"
 install-shared-lib:
 	$(INSTALL) -d "$(DESTDIR)$(LIBDIR)"
 	$(INSTALL) -m 755 $(LIBGIFSO) "$(DESTDIR)$(LIBDIR)/$(LIBGIFSOVER)"
 	ln -sf $(LIBGIFSOVER) "$(DESTDIR)$(LIBDIR)/$(LIBGIFSOMAJOR)"
 	ln -sf $(LIBGIFSOMAJOR) "$(DESTDIR)$(LIBDIR)/$(LIBGIFSO)"
+	$(INSTALL) -m 755 libgifutil.so "$(DESTDIR)$(LIBDIR)/libgifutil.so.$(LIBVER)"
+	ln -sf libgifutil.so.$(LIBVER) "$(DESTDIR)$(LIBDIR)/libgifutil.so.$(LIBMAJOR)"
+	ln -sf libgifutil.so.$(LIBMAJOR) "$(DESTDIR)$(LIBDIR)/libgifutil.so"
 install-lib: install-static-lib install-shared-lib
 install-man:
 	$(INSTALL) -d "$(DESTDIR)$(MANDIR)/man1" "$(DESTDIR)$(MANDIR)/man7"
@@ -206,6 +211,8 @@ uninstall-include:
 uninstall-lib:
 	cd "$(DESTDIR)$(LIBDIR)" && \
 		rm -f libgif.a $(LIBGIFSO) $(LIBGIFSOMAJOR) $(LIBGIFSOVER)
+	cd "$(DESTDIR)$(LIBDIR)" && \
+		rm -f libgifutil.a libgifutil.so libgifutil.so.$(LIBMAJOR) libgifutil.so.$(LIBVER)
 uninstall-man:
 	cd "$(DESTDIR)$(MANDIR)/man1" && rm -f $(shell cd doc >/dev/null && echo *.1)
 	cd "$(DESTDIR)$(MANDIR)/man7" && rm -f $(shell cd doc >/dev/null && echo *.7)
