--- giflib-5.1.9/Makefile.orig	2019-04-26 17:33:39.313133728 +0200
+++ giflib-5.1.9/Makefile	2019-04-26 17:35:09.355979257 +0200
@@ -60,12 +60,12 @@
 	gifsponge \
 	gifwedge
 
-LDLIBS=libgif.a -lm
+LDLIBS=-L. -lgif -lm
 
 all: libgif.so libgif.a libutil.so libutil.a $(UTILS)
 	$(MAKE) -C doc
 
-$(UTILS):: libgif.a libutil.a
+$(UTILS):: libgif.so libutil.a
 
 libgif.so: $(OBJECTS) $(HEADERS)
 	$(CC) $(CFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,libgif.so.$(LIBMAJOR) -o libgif.so $(OBJECTS)
