--- giflib-5.2.2/Makefile.orig	2024-02-25 21:33:25.080972300 +0100
+++ giflib-5.2.2/Makefile	2024-02-25 21:50:41.035360053 +0100
@@ -61,7 +61,7 @@ UTILS = $(INSTALLABLE) \
 	gifsponge \
 	gifwedge
 
-LDLIBS=libgif.a -lm
+LDLIBS=-L. -lgif -lm
 
 MANUAL_PAGES = \
 	doc/gif2rgb.xml \
@@ -76,24 +76,24 @@ SOEXTENSION	= so
 LIBGIFSO	= libgif.$(SOEXTENSION)
 LIBGIFSOMAJOR	= libgif.$(SOEXTENSION).$(LIBMAJOR)
 LIBGIFSOVER	= libgif.$(SOEXTENSION).$(LIBVER)
-LIBUTILSO	= libutil.$(SOEXTENSION)
-LIBUTILSOMAJOR	= libutil.$(SOEXTENSION).$(LIBMAJOR)
+LIBUTILSO	= libgifutil.$(SOEXTENSION)
+LIBUTILSOMAJOR	= libgifutil.$(SOEXTENSION).$(LIBMAJOR)
 ifeq ($(UNAME), Darwin)
 SOEXTENSION	= dylib
 LIBGIFSO        = libgif.$(SOEXTENSION)
 LIBGIFSOMAJOR   = libgif.$(LIBMAJOR).$(SOEXTENSION)
 LIBGIFSOVER	= libgif.$(LIBVER).$(SOEXTENSION)
-LIBUTILSO	= libutil.$(SOEXTENSION)
-LIBUTILSOMAJOR	= libutil.$(LIBMAJOR).$(SOEXTENSION)
+LIBUTILSO	= libgifutil.$(SOEXTENSION)
+LIBUTILSOMAJOR	= libgifutil.$(LIBMAJOR).$(SOEXTENSION)
 endif
 
-all: $(LIBGIFSO) libgif.a $(LIBUTILSO) libutil.a $(UTILS)
+all: $(LIBGIFSO) libgif.a $(LIBUTILSO) libgifutil.a $(UTILS)
 ifeq ($(UNAME), Darwin)
 else
 	$(MAKE) -C doc
 endif
 
-$(UTILS):: libgif.a libutil.a
+$(UTILS):: $(LIBGIFSO) $(LIBUTILSO)
 
 $(LIBGIFSO): $(OBJECTS) $(HEADERS)
 ifeq ($(UNAME), Darwin)
@@ -105,18 +105,18 @@ endif
 libgif.a: $(OBJECTS) $(HEADERS)
 	$(AR) rcs libgif.a $(OBJECTS)
 
-$(LIBUTILSO): $(UOBJECTS) $(UHEADERS)
+$(LIBUTILSO): $(UOBJECTS) $(UHEADERS) libgif.so
 ifeq ($(UNAME), Darwin)
 	$(CC) $(CFLAGS) -dynamiclib -current_version $(LIBVER) $(OBJECTS) -o $(LIBUTILSO)
 else
-	$(CC) $(CFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,$(LIBUTILMAJOR) -o $(LIBUTILSO) $(UOBJECTS)
+	$(CC) $(CFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,$(LIBUTILSOMAJOR) -o $(LIBUTILSO) $(UOBJECTS) -L. -lgif
 endif
 
-libutil.a: $(UOBJECTS) $(UHEADERS)
-	$(AR) rcs libutil.a $(UOBJECTS)
+libgifutil.a: $(UOBJECTS) $(UHEADERS)
+	$(AR) rcs libgifutil.a $(UOBJECTS)
 
 clean:
-	rm -f $(UTILS) $(TARGET) libgetarg.a libgif.a $(LIBGIFSO) libutil.a $(LIBUTILSO) *.o
+	rm -f $(UTILS) $(TARGET) libgetarg.a libgif.a $(LIBGIFSO) libgifutil.a $(LIBUTILSO) *.o
 	rm -f $(LIBGIFSOVER)
 	rm -f $(LIBGIFSOMAJOR)
 	rm -fr doc/*.1 *.html doc/staging
@@ -141,12 +141,17 @@ install-bin: $(INSTALLABLE)
 install-include:
 	$(INSTALL) -d "$(DESTDIR)$(INCDIR)"
 	$(INSTALL) -m 644 gif_lib.h "$(DESTDIR)$(INCDIR)"
+	$(INSTALL) -m 644 getarg.h "$(DESTDIR)$(INCDIR)/gif_util.h"
 install-lib:
 	$(INSTALL) -d "$(DESTDIR)$(LIBDIR)"
 	$(INSTALL) -m 644 libgif.a "$(DESTDIR)$(LIBDIR)/libgif.a"
 	$(INSTALL) -m 755 $(LIBGIFSO) "$(DESTDIR)$(LIBDIR)/$(LIBGIFSOVER)"
 	ln -sf $(LIBGIFSOVER) "$(DESTDIR)$(LIBDIR)/$(LIBGIFSOMAJOR)"
 	ln -sf $(LIBGIFSOMAJOR) "$(DESTDIR)$(LIBDIR)/$(LIBGIFSO)"
+	$(INSTALL) -m 644 libgifutil.a "$(DESTDIR)$(LIBDIR)/libgifutil.a"
+	$(INSTALL) -m 755 libgifutil.so "$(DESTDIR)$(LIBDIR)/libgifutil.so.$(LIBVER)"
+	ln -sf libgifutil.so.$(LIBVER) "$(DESTDIR)$(LIBDIR)/libgifutil.so.$(LIBMAJOR)"
+	ln -sf libgifutil.so.$(LIBMAJOR) "$(DESTDIR)$(LIBDIR)/libgifutil.so"
 install-man:
 	$(INSTALL) -d "$(DESTDIR)$(MANDIR)/man1"
 	$(INSTALL) -m 644 $(MANUAL_PAGES) "$(DESTDIR)$(MANDIR)/man1"
@@ -158,6 +163,8 @@ uninstall-include:
 uninstall-lib:
 	cd "$(DESTDIR)$(LIBDIR)" && \
 		rm -f libgif.a $(LIBGIFSO) $(LIBGIFSOMAJOR) $(LIBGIFSOVER)
+	cd "$(DESTDIR)$(LIBDIR)" && \
+		rm -f libgifutil.a libgifutil.so libgifutil.so.$(LIBMAJOR) libgifutil.so.$(LIBVER)
 uninstall-man:
 	cd "$(DESTDIR)$(MANDIR)/man1" && rm -f $(shell cd doc >/dev/null && echo *.1)
 	cd "$(DESTDIR)$(MANDIR)/man7" && rm -f $(shell cd doc >/dev/null && echo *.7)
