*** giftext.c.orig	2006-08-21 15:41:47.000000000 -0400
--- giftext.c	2006-08-21 15:41:55.000000000 -0400
***************
*** 135,141 ****
  	       GifFileName, GifFile->SWidth, GifFile->SHeight);
  	printf("\tColorResolution = %d, BitsPerPixel = %d, BackGround = %d.\n",
  	       GifFile->SColorResolution,
! 	       GifFile->SColorMap->BitsPerPixel,
  	       GifFile->SBackGroundColor);
  	if (GifFile->SColorMap)
  	    printf("\tHas Global Color Map.\n\n");
--- 135,141 ----
  	       GifFileName, GifFile->SWidth, GifFile->SHeight);
  	printf("\tColorResolution = %d, BitsPerPixel = %d, BackGround = %d.\n",
  	       GifFile->SColorResolution,
! 	       GifFile->SColorMap?GifFile->SColorMap->BitsPerPixel:0,
  	       GifFile->SBackGroundColor);
  	if (GifFile->SColorMap)
  	    printf("\tHas Global Color Map.\n\n");
